edn-api/routes/routes.go

```diff
@@ -9,11 +10,15 @@ import (
 
 func InitializeRoutes(r *gin.Engine, db *database.StatementRepository) {
 
-	api := r.Group("/api") 
+	api := r.Group("/api")
 	{
 		api.GET("/getListStatement/:id", handlers.GetListStatementById(db))
 		api.GET("/getAllListStatements", handlers.GetAllListStatements(db))
 		api.GET("/getAllImageStatements", handlers.GetAllImageStatements(db))
-		api.GET("/testOpenAI", handlers.TestOpenAI())
+
+	}
+	protected := r.Group("/api", auth.AuthenticateWithClerk())
+	{
+		protected.GET("/testOpenAI", handlers.TestOpenAI())
```

Comment:
We can pretty simply just add routes that need to be authenticated into this separate api group. Anything that is listed in here requires the AuthenticateWithClerk check to pass in order to reach the handlers.